pdns/mastercommunicator.cc

```diff
@@ -93,7 +95,7 @@ void CommunicatorClass::queueNotifyDomain(const DNSName &domain, UeberBackend *B
     }
   }
 
-  if (!hasQueuedItem)
+  if (!hasQueuedItem && d_onlyNotify.size())
```

Comment:
You are "fixing" a very specific re-notify case here. But you are hiding  a warning which also applies to master domains. If only-notify is empty they rely on also-notify to make things work. The warning is here to detect misconfigured master and slave-renotify domains. 
Yes, it will also warn for slave domains with an empty notification list, if global slave-renotify is enabled. But this is something you can't avoid without hiding real problems in other usecases.
In my opinion the only real fix, for your usecase, is add a SLAVE-RENOTIFY metadata option, so it is possible to enable renotification on a per domain basis. 

@Habbie a startup warning is useless because a part of the notify logic is stored in the domain metadata. Empty only-notify does not say a thing at startup.